### PR TITLE
Update to latest core

### LINKED ISSUE #7775: [Realm compact] - io.realm.exceptions.RealmFileException: Unable to open a realm at path 'xxx.realm' Realm file initial open failed: Invalid streaming format cookie (0) Path:Exception backtrace: <backtrace not supported on this platform>
### How frequently does the bug occur?

Sometimes

### Description

Hi,

I have the same issue than https://github.com/realm/realm-java/issues/4237 for the same same problem. 

I have 32.000 crashes for 50 users (for 250.000 users in the month)
image

I don't find any solution, can you help me please ? 

### Stacktrace & log output

```shell
Caused by io.realm.exceptions.RealmFileException
Unable to open a realm at path '/data/user/0/com.app.androidfr/files/config.realm': Realm file initial open failed: Invalid streaming format cookie (0) Path:Exception backtrace: <backtrace not supported on this platform> Path: /data/user/0/com.app.androidfr/files/config.realm Exception backtrace: <backtrace not supported on this platform>. (Realm file initial open failed: Invalid streaming format cookie (0) Path: /data/user/0/com.app.androidfr/files/config.realm Exception backtrace: <backtrace not supported on this platform> Path: /data/user/0/com.app.androidfr/files/config.realm Exception backtrace: <backtrace not supported on this platform>) (/data/user/0/com.app.androidfr/files/config.realm) in /tmp/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_OsSharedRealm.cpp line 107

io.realm.internal.OsSharedRealm.nativeGetSharedRealm (OsSharedRealm.java)
io.realm.internal.OsSharedRealm.<init> (OsSharedRealm.java:174)
io.realm.internal.OsSharedRealm.getInstance (OsSharedRealm.java:259)
io.realm.internal.OsSharedRealm.getInstance (OsSharedRealm.java:247)
io.realm.BaseRealm.compactRealm (BaseRealm.java:814)
io.realm.Realm.compactRealm (Realm.java:1954)
com.app.helper.realm.ConfigModule$Companion.compact (ConfigModule.kt:47)
com.app.App.initRealm (App.kt:70)
com.app.App.onCreate (App.kt:33)
```

### Can you reproduce the bug?

No

### Reproduction Steps

_No response_

### Version

10.13.0

### Are you using encryption?

No

### Platform OS and version(s)

Android

### Build environment

Android Build Tools version: 7.6-bin.zip
Gradle version: 7.4.1

### PR DESCRIPTION
**Pull request by rorbech**
Closes #7775

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsAppCredentials.java [hunk 1]
```diff
@@ -51,10 +50,6 @@ public static OsAppCredentials apiKey(String key) {
         return new OsAppCredentials(nativeCreate(TYPE_API_KEY, key));
     }
 
-    public static OsAppCredentials serverApiKey(String key) {
```
Comment (rorbech) [inline realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsAppCredentials.java:54]: This wasn't really used and not exposed in the SDK. Removed from core in https://github.com/realm/realm-core/issues/5914

**Review by clementetb — APPROVED** [review]
LGTM

**Review by cmelchior — APPROVED** [review]

**Comment by cmelchior** [comment]
Ready for review

File: realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java [hunk 2]
```diff
@@ -862,7 +861,8 @@ public void isNotEmpty_acrossLinkingObjectObjectLink() {
                     assertEquals(0, realm.where(AllJavaTypesUnsupportedTypes.class).isNotEmpty(AllJavaTypesUnsupportedTypes.FIELD_LO_OBJECT + "." + AllJavaTypesUnsupportedTypes.FIELD_REALM_ANY_SET).count());
                     break;
                 default:
-                    fail("Unknown type: " + type);
+                    // Do nothing
+                    // fail("Unknown type: " + type);
```
Comment (nhachicha) [inline realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java:865]: should we keep failing ?

File: realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java [hunk 3]
```diff
@@ -401,11 +400,11 @@ public void isEmpty_acrossLink() {
                 case LINKING_OBJECTS:
                     // Rows 0 and 1 are not included as they are linked to another row through FIELD_OBJECT
                     // Row 2 is included (no link)
-                    assertEquals(1, realm.where(AllJavaTypesUnsupportedTypes.class).isEmpty(AllJavaTypesUnsupportedTypes.FIELD_OBJECT + "." + AllJavaTypesUnsupportedTypes.FIELD_LO_OBJECT).count());
```
Comment (nhachicha) [inline realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java:404]: Why the expected values changed for these tests?

File: realm/realm-library/src/main/cpp/io_realm_mongodb_sync_SyncSession.cpp [hunk 4]
```diff
@@ -69,10 +69,8 @@ void handleCompletion(jint callback_id, JavaMethod &java_notify_result_method,
     if (!status.is_ok()) {
         jbyte category = categoryAsJByte(status);
         java_error_category = JavaLocalRef<jobject>(env, JavaClassGlobalDef::new_long(env,category));
-        java_error_code = JavaLocalRef<jobject>(env, JavaClassGlobalDef::new_long(env,
-                                                                                  status.get_std_error_code().value()));
-        java_error_message = JavaLocalRef<jstring>(env, env->NewStringUTF(
-                status.reason().c_str()));
+        java_error_code = JavaLocalRef<jobject>(env, JavaClassGlobalDef::new_long(env, static_cast<int32_t>(status.code())));
+        java_error_message = JavaLocalRef<jstring>(env, env->NewStringUTF( status.reason().c_str()));
```
Comment (nhachicha) [inline realm/realm-library/src/main/cpp/io_realm_mongodb_sync_SyncSession.cpp:73]: maybe use `to_jstring`?

File: realm/realm-library/src/main/cpp/util.cpp [hunk 5]
```diff
@@ -15,6 +15,7 @@
  */
 
 #include <algorithm>
+#include <iomanip>
```
Comment (nhachicha) [inline realm/realm-library/src/main/cpp/util.cpp:18]: left over?

File: realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java (hunk 3 above)
Comment (cmelchior) [reply realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java:404]: It was changed after https://github.com/realm/realm-core/pull/6867. There was a lot of discussion about this, but in the end, we determined that the old behavior was buggy and subsequently fixed.

**Pull request merged**